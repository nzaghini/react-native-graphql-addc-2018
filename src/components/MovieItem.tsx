import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { ALL_MOVIES_QUERY_movies as Movie } from "../queries/models/ALL_MOVIES_QUERY";

interface MovieProp {
    movie: Movie;
    key: string;
}

const MovieItem = (props: MovieProp) => {
    const { container, titleStyle, subTitleStyle } = styles;
    const { title, year, director } = props.movie;
    return (
        <View style={container}>
            <Text style={titleStyle}>{title} - {year}</Text>
            <Text style={subTitleStyle}>{director.firstName} {director.lastName}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        height: 60,
        justifyContent: "center",
        paddingLeft: 10,
        borderColor: "#ddd",
        borderBottomWidth: 1,
    },
    titleStyle: {
        fontSize: 18,
    },
    subTitleStyle: {
        fontSize: 12,
    },
});

export default MovieItem;
