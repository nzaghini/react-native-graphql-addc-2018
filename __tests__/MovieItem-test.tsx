import React from 'react';
import renderer from 'react-test-renderer';
import MovieItem from '../src/components/MovieItem';

const mockListOfMovies = [ 
    { id: 1, title: 'Interstellar', year: 2014, director: { firstName: 'Christopher', lastName: 'Nolan' } },
    { id: 2, title: 'Mad Max: Fury Road', year: 2015, director: { firstName: 'George', lastName: 'Miller' } }
];

describe('<MovieItem />', () => {
    it('renders correctly first movie', () => {
        const tree = renderer.create(<MovieItem key="1" movie={mockListOfMovies[0]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly second movie', () => {
        const tree = renderer.create(<MovieItem key="1" movie={mockListOfMovies[1]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
