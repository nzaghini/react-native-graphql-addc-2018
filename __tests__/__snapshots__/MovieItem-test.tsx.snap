// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItem /> renders correctly first movie 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="movieId_1"
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "height": 60,
        "justifyContent": "center",
        "paddingLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Interstellar
       - 
      2014
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Christopher
       
      Nolan
    </Text>
  </View>
</View>
`;

exports[`<MovieItem /> renders correctly second movie 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="movieId_2"
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "height": 60,
        "justifyContent": "center",
        "paddingLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Mad Max: Fury Road
       - 
      2015
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      George
       
      Miller
    </Text>
  </View>
</View>
`;
